frontend/src/components/SignUpPage.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {ChangeEvent, FormEvent, useState} from "react";
+import axios from "axios";
+import {
+    Box,
+    Button,
+    Container,
+    FormControl,
+    InputLabel,
+    Select,
+    SelectChangeEvent,
+    Snackbar,
+    TextField,
+    Typography
+} from "@mui/material";
+import Layout from "./Layout";
+import useAuth from "../hooks/useAuth";
+import MenuItem from "@mui/material/MenuItem";
+
+export default function SignUp() {
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [role, setRole] = useState("");
+    const [confirmPassword, setConfirmPassword] = useState("");
+    const [error, setError] = useState("");
+    const {user} = useAuth(true);
+    const isAdmin = user?.role === "admin";
+
+    const [openSnackbar, setOpenSnackbar] = useState(false);
+    const handleOpenSnackbar = () => {
+        setOpenSnackbar(true);
+    };
+
+    function handleUsernameChange(event: ChangeEvent<HTMLInputElement>) {
+        setUsername(event.target.value);
+    }
+
+    function handlePasswordChange(event: ChangeEvent<HTMLInputElement>) {
+        setPassword(event.target.value);
+    }
+
+    function handleConfirmPasswordChange(event: ChangeEvent<HTMLInputElement>) {
+        setConfirmPassword(event.target.value);
+    }
+
+    const handleRoleChange = (event: SelectChangeEvent) => {
+        setRole(event.target.value as string);
+    }
+
+    function handleSubmit(event: FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+
+        if (!isAdmin) {
+            setError("Nur Administratoren dürfen neue Nutzer registrieren.")
+        }
+
+        if (password !== confirmPassword) {
+            setError("Passwörter stimmen nicht überein.");
+            return;
+        }
+
+        axios
+            .post("/api/user", {username, password, role})
+            .then(() => {
+                handleOpenSnackbar()
+                handleOpenSnackbar()
+                setUsername("")
+                setPassword("")
+                setConfirmPassword("")
+                setRole("")
+            })
+            .catch((error) =>
+                setError(
+                    error.response?.data?.message || "Registrierung fehlgeschlagen."
+                )
+            );
```

Comment:
Maybe you should create a new directory in your frontend ( for example "hooks") to save your axios request and so , you only need here to make its call